src/test/java/org/cactoos/experimental/ThreadsTest.java

```diff
@@ -25,65 +25,103 @@
 package org.cactoos.experimental;
 
 import java.io.UncheckedIOException;
+import java.time.Duration;
+import java.util.concurrent.CancellationException;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
-import org.cactoos.Proc;
-import org.cactoos.func.Repeated;
-import org.cactoos.func.UncheckedFunc;
-import org.cactoos.scalar.ScalarOf;
+import org.junit.jupiter.api.RepeatedTest;
 import org.junit.jupiter.api.Test;
 import org.llorllale.cactoos.matchers.Assertion;
 import org.llorllale.cactoos.matchers.HasValues;
 import org.llorllale.cactoos.matchers.Throws;
+import org.llorllale.cactoos.shaded.org.cactoos.scalar.LengthOf;
 
 /**
  * Test case for {@link Threads}.
  *
  * @since 1.0.0
- * @checkstyle MagicNumberCheck (500 lines)
  * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 lines)
  */
 @SuppressWarnings("PMD.AvoidDuplicateLiterals")
 final class ThreadsTest {
 
+    /**
+     * Repetitions when running test several times.
+     */
+    private static final int REPETITIONS = 5;
+
     /**
      * Execute the tasks concurrently using {@link Threads} when
      *  {@link ExecutorService} was initiated by someone else.
      */
-    @Test
-    void containsResults() {
-        this.repeat(
-            arg -> {
-                final ExecutorService extor = Executors.newFixedThreadPool(3);
-                try {
-                    new Assertion<>(
-                        "contains results from callables",
-                        new Threads<String>(
-                            extor,
-                            () -> {
-                                this.sleep();
-                                return "txt 1";
-                            },
-                            () -> {
-                                this.sleep();
-                                return "txt 2";
-                            },
-                            () -> {
-                                this.sleep();
-                                return "txt 3";
-                            }
-                        ),
-                        new HasValues<>("txt 1", "txt 2", "txt 3")
-                    ).affirm();
-                } finally {
-                    extor.shutdown();
-                    if (!extor.awaitTermination(1L, TimeUnit.SECONDS)) {
-                        extor.shutdownNow();
+    @RepeatedTest(ThreadsTest.REPETITIONS)
+    void containsResults() throws Exception {
+        final ExecutorService extor = Executors.newFixedThreadPool(3);
+        try {
+            new Assertion<>(
+                "Must contain results from callables",
+                new Threads<String>(
+                    extor,
+                    Duration.ofSeconds(1),
+                    () -> {
```

Comment:
@victornoel How about declaring three callables property of `ThreadsTest` (`firsttext`, `secondtext`, `thirdtext`) and initializing them in setUp method ? I notice that it is repeated in each test.

Reply:
@baudoliver7 it  is a good practice not to share knowledge between tests, it makes them more independent and more resilent, see https://www.yegor256.com/2016/05/03/test-methods-must-share-nothing.html for a related argument :)

Reply:
@victornoel Good !